src/sentry/static/sentry/app/components/events/interfaces/breadcrumbs/breadcrumbs.tsx

```diff
@@ -0,0 +1,220 @@
+
+    const exception = event.entries.find(entry => entry.type === 'exception');
+
+    if (!exception && !event.message) {
+      return undefined;
+    }
+
+    if (exception) {
+      const {type, value, module: mdl} = exception.data.values[0];
+      return {
+        type: 'error',
+        level: 'error',
+        category: this.moduleToCategory(mdl) || 'exception',
+        data: {
+          type,
+          value,
+        },
+        timestamp: event.dateCreated,
+      };
+    }
+
+    const levelTag = (event.tags || []).find(tag => tag.key === 'level');
+
+    return {
+      type: 'message',
+      level: levelTag?.value as Crumb['level'],
+      category: 'message',
+      message: event.message,
+      timestamp: event.dateCreated,
+    };
+  };
+
+  getCollapsedCrumbQuantity = (): {
+    filteredCollapsedCrumbs: Array<Crumb>;
+    collapsedQuantity: number;
+  } => {
+    const {isCollapsed, filteredCrumbs} = this.state;
+
+    let filteredCollapsedCrumbs = filteredCrumbs;
+
+    if (isCollapsed && filteredCollapsedCrumbs.length > MAX_CRUMBS_WHEN_COLLAPSED) {
+      filteredCollapsedCrumbs = filteredCollapsedCrumbs.slice(-MAX_CRUMBS_WHEN_COLLAPSED);
+    }
+
+    return {
+      filteredCollapsedCrumbs,
+      collapsedQuantity: filteredCrumbs.length - filteredCollapsedCrumbs.length,
+    };
+  };
+
+  handleChangeSearchTerm = (searchTerm: string) => {
+    const {crumbs} = this.state;
+
+    const filteredCrumbs = crumbs.filter(
+      item =>
+        // return true if any of category, message, or level contain queryValue
+        !!['category', 'message', 'level'].find(prop => {
+          const propValue = (item[prop] || '').toLowerCase();
+          return propValue.includes(searchTerm);
+        })
+    );
+
+    this.setState({
+      searchTerm,
+      filteredCrumbs,
+    });
+  };
+
+  handleCollapseToggle = () => {
+    this.setState(prevState => ({
+      isCollapsed: !prevState.isCollapsed,
+    }));
+  };
+
+  handleCleanSearch = () => {
+    this.setState({
+      searchTerm: '',
+      isCollapsed: true,
+    });
+  };
+
+  render() {
+    const {event, type} = this.props;
+    const {searchTerm} = this.state;
+    const {collapsedQuantity, filteredCollapsedCrumbs} = this.getCollapsedCrumbQuantity();
+
+    return (
+      <EventDataSection
+        type={type}
+        title={
+          <h3>
+            <GuideAnchor target="breadcrumbs" position="bottom">
+              {t('Breadcrumbs')}
+            </GuideAnchor>
+          </h3>
+        }
+        actions={
+          <BreadCrumbsSearch
+            searchTerm={searchTerm}
+            onChangeSearchTerm={this.handleChangeSearchTerm}
+            onClearSearchTerm={this.handleCleanSearch}
+          />
+        }
+        wrapTitle={false}
+      >
+        <Content>
+          {filteredCollapsedCrumbs.length > 0 ? (
+            <PlatformContextProvider value={{platform: event.platform}}>
+              <BreadCrumbs className="crumbs">
```

Comment:
Do we still need this className?

Reply:
Yes we unfortunately do... otherwise I would have to refactor more components than I actually wanted in this PR